cborg/src/Codec/CBOR/Read.hs

```diff
@@ -785,31 +827,39 @@ go_slow_overlapped da sz bs_cur bs_next !offset =
         assert (BS.null bs_empty) $
         return (bs', offset', x)
 
-      SlowConsumeTokenBytes bs_empty k len ->
-        assert (BS.null bs_empty) $ do
-        (bstr, bs'') <- if BS.length bs' < len
-                          then getTokenVarLen len bs' offset'
-                          else let !bstr = BS.take len bs'
-                                   !bs'' = BS.drop len bs'
-                                in return (bstr, bs'')
-        lift (k bstr) >>= \daz -> go_slow daz bs'' (offset' + fromIntegral len)
+      SlowConsumeTokenBytes bs_empty k len         -> consume id                bs' offset' bs_empty k len
+
+      SlowConsumeTokenByteArray bs_empty k len     -> consume BA.fromByteString bs' offset' bs_empty k len
+
+      SlowConsumeTokenString bs_empty k len        -> consume T.decodeUtf8      bs' offset' bs_empty k len
+        -- TODO FIXME: utf8 validation
 
-      SlowConsumeTokenString bs_empty k len ->
+      SlowConsumeTokenUtf8ByteArray bs_empty k len -> consume BA.fromByteString bs' offset' bs_empty k len
+        -- TODO FIXME: utf8 validation
```

Comment:
Right, this doesn't validate at all, at least the T.decodeUTF8 will validate but it throws an exception rather than failing within the monad.

Reply:
It's not entirely clear to me whether we even *should* validate in this case. I am leaning towards not validating (and clearly documenting this).

Reply:
Right. I'm happy with documenting that we do not validate the UTF8, in which case these TODOs are redundant. The one for `Text` is still valid, since we should be properly handling the decoding error in that case.